packages/network-controller/src/NetworkController.test.ts

```diff
@@ -3722,6 +3724,108 @@ describe('NetworkController', () => {
+      await withController({ messenger }, async ({ controller }) => {
+        const rpcUrlNetwork = {
+          chainId: undefined,
+          nickname: 'RPC',
+          rpcPrefs: undefined,
+          rpcUrl: 'rpc_url',
+          ticker: 'RPC',
+        };
+        const localhostNetwork = {
+          chainId: undefined,
+          nickname: undefined,
+          rpcPrefs: undefined,
+          rpcUrl: 'http://localhost:8545',
+          ticker: 'LOCAL',
+        };
+        controller.upsertNetworkConfiguration(rpcUrlNetwork);
+        controller.upsertNetworkConfiguration(localhostNetwork);
+
+        expect(
+          Object.values(controller.state.networkConfigurations),
+        ).toStrictEqual(
+          expect.arrayContaining([rpcUrlNetwork, localhostNetwork]),
+        );
+        controller.upsertNetworkConfiguration({ rpcUrl: 'rpc_url' });
+
+        expect(
+          Object.values(controller.state.networkConfigurations),
+        ).toStrictEqual(
+          expect.arrayContaining([
+            localhostNetwork,
+            { ...rpcUrlNetwork, nickname: undefined, ticker: undefined },
+          ]),
+        );
+      });
+    });
+
+    it('should update a network configuration when the configuration being added has an rpcURL that matches an existing configuration', async () => {
+      const messenger = buildMessenger();
+      await withController({ messenger }, async ({ controller }) => {
+        controller.upsertNetworkConfiguration({
+          rpcUrl: 'test_rpc_url',
+          ticker: 'old_rpc_ticker',
+          nickname: 'old_rpc_nickname',
+        });
+
+        expect(
+          Object.values(controller.state.networkConfigurations),
+        ).toStrictEqual([
+          expect.objectContaining({
+            rpcUrl: 'test_rpc_url',
+            nickname: 'old_rpc_nickname',
+            ticker: 'old_rpc_ticker',
+            rpcPrefs: undefined,
+            chainId: undefined,
+          }),
+        ]);
+
+        controller.upsertNetworkConfiguration({
+          rpcUrl: 'test_rpc_url',
+          ticker: 'new_rpc_ticker',
+          nickname: 'new_rpc_nickname',
+        });
+
+        expect(
+          Object.values(controller.state.networkConfigurations),
+        ).toStrictEqual([
+          expect.objectContaining({
+            rpcUrl: 'test_rpc_url',
+            nickname: 'new_rpc_nickname',
+            ticker: 'new_rpc_ticker',
+            rpcPrefs: undefined,
+            chainId: undefined,
+          }),
+        ]);
+      });
+    });
+
+    it('should remove a network configuration', async () => {
+      const messenger = buildMessenger();
+      await withController({ messenger }, async ({ controller }) => {
+        const rpcUrlNetwork = {
+          chainId: '0x123',
+          nickname: 'test_rpc_url',
+          rpcPrefs: { blockExplorerUrl: 'test_block_explorer' },
+          rpcUrl: 'rpc_url',
+          ticker: 'TEST',
+        };
+        const uuid_rpc_url =
```

Comment:
Do we want to call this `uuidRpcUrl`?

Reply:
done [here](https://github.com/MetaMask/core/pull/1064/commits/ac392e90922842cc69a7cb6a7377557c532dbc35) => now called `testUUID`